cpp/modmesh/io/gmsh.hpp

```diff
@@ -0,0 +1,397 @@
+        MM_DECL_SWITCH_ELEMENT_TYPE(13,   3,     18,         7, VEC(0, 2, 1, 3, 5, 4))           // 18-node prism
+        MM_DECL_SWITCH_ELEMENT_TYPE(14,   3,     14,         8, VEC(0, 1, 2, 3, 4))              // 14-node pyramid
+        MM_DECL_SWITCH_ELEMENT_TYPE(15,   0,      1,         1, VEC(0))                          // 1-node point
+        MM_DECL_SWITCH_ELEMENT_TYPE(16,   2,      8,         3, VEC(0, 1, 2, 3))                 // 8-node quadrangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(17,   3,     20,         5, VEC(0, 1, 2, 3, 4, 5, 6, 7))     // 20-node hexahedron
+        MM_DECL_SWITCH_ELEMENT_TYPE(18,   3,     15,         7, VEC(0, 2, 1, 3, 5, 4))           // 15-node prism
+        MM_DECL_SWITCH_ELEMENT_TYPE(19,   3,     13,         8, VEC(0, 1, 2, 3, 4))              // 13-node pyramid
+        MM_DECL_SWITCH_ELEMENT_TYPE(20,   2,      9,         4, VEC(0, 1, 2))                    // 9-node incomplete triangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(21,   2,     10,         4, VEC(0, 1, 2))                    // 10-node triangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(22,   2,     12,         4, VEC(0, 1, 2))                    // 12-node incomplete triangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(23,   2,     15,         4, VEC(0, 1, 2))                    // 15-node triangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(24,   2,     15,         4, VEC(0, 1, 2))                    // 15-node incomplete triangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(25,   2,     21,         4, VEC(0, 1, 2))                    // 21-node incomplete triangle
+        MM_DECL_SWITCH_ELEMENT_TYPE(26,   1,      4,         2, VEC(0, 1))                       // 4-node edge
+        MM_DECL_SWITCH_ELEMENT_TYPE(27,   1,      5,         2, VEC(0, 1))                       // 5-node edge
+        MM_DECL_SWITCH_ELEMENT_TYPE(28,   1,      6,         2, VEC(0, 1))                       // 6-node edge
+        MM_DECL_SWITCH_ELEMENT_TYPE(29,   3,     20,         6, VEC(0, 1, 2, 3))                 // 20-node tetrahedron
+        MM_DECL_SWITCH_ELEMENT_TYPE(30,   3,     35,         6, VEC(0, 1, 2 ,3))                 // 35-node tetrahedron
+        MM_DECL_SWITCH_ELEMENT_TYPE(31,   3,     56,         6, VEC(0, 1, 2, 3))                 // 56-node tetrahedron
+        MM_DECL_SWITCH_ELEMENT_TYPE(92,   3,     64,         5, VEC(0, 1, 2, 3, 4, 5, 6, 7))     // 64-node hexahedron
+        MM_DECL_SWITCH_ELEMENT_TYPE(93,   3,    125,         5, VEC(0, 1, 2, 3, 4, 5, 6, 7))     // 125-node hexahedron
+        default: return ElementDef{}; break;
+        // clang-format on
+    }
+#undef MM_DECL_SWITCH_ELEMENT_TYPE
+#undef VEC
+}
+
+class Gmsh
+    : public NumberBase<int32_t, double>
+{
+    using number_base = NumberBase<int32_t, double>;
+    using int_type = typename number_base::int_type;
+    using uint_type = typename number_base::uint_type;
+    using real_type = typename number_base::real_type;
+
+public:
+    explicit Gmsh(const std::string & file_path)
```

Comment:
Could you make it also take a string as data, not a string as path?  Both are useful.

Reply:
I change it to use string as data and python can take care of file stream to `bytes`, even if user is using `Gmsh` in cpp, cpp also can take care of file stream to string.